include/devicetree/gpio.h

```diff
@@ -54,7 +54,8 @@ extern "C" {
  * @see DT_PHANDLE_BY_IDX()
  */
 #define DT_GPIO_LABEL_BY_IDX(node_id, gpio_pha, idx) \
-	DT_PROP_BY_PHANDLE_IDX(node_id, gpio_pha, idx, label)
+	UTIL_AND(DT_HAS_PROP(DT_PHANDLE_BY_IDX(node_id, gpio_pha, idx), label), \
```

Comment:
This will return a 0 (and therefore NULL) label if there is no property. Is that what you want?